go/mysql/replication_position.go

```diff
@@ -169,3 +169,31 @@ func (rp *Position) UnmarshalJSON(buf []byte) error {
 	}
 	return nil
 }
+
+// Comparable returns whether the receiver is comparable to the supplied position, based on whether one
+// of the two positions contains the other.
+func (rp *Position) Comparable(other Position) bool {
+	if !rp.GTIDSet.Contains(other.GTIDSet) && !other.GTIDSet.Contains(rp.GTIDSet) {
+		return false
+	}
+
+	return true
```

Comment:
Alternatively:

```suggestion
	return rp.GTIDSet.Contains(other.GTIDSet) || other.GTIDSet.Contains(rp.GTIDSet)
```

Also, do all the Contains() implementations do the right thing when either the receiver or the argument are nil?

Reply:
Good suggestion - updated. I just checked. Here's the rundown:
MariaDB -  When other `GTIDSet` is nil, we return true, but we don't bother to check if the receiver is nil at all. 
MySQL - Doesn't nil check receiver or other at all.
FilePos - When other `GTIDSet` is nil, we return true, but never check the receiver (same as MariaDB)

Reply:
Fixing that now.

Reply:
Fixed.